raspberry-pi/4/pkgs-overlays.nix

```
@@ -0,0 +1,22 @@
+{ config, lib, pkgs, modulesPath, ... }:
+let
+  callPackage = path: overrides:
+    let f = import path;
+    in f ((builtins.intersectAttrs (builtins.functionArgs f) pkgs) // overrides);
+  cfg = config.hardware.raspberry-pi."4".apply-overlays-dtmerge;
+  dt_ao_overlay = (final: prev: {
+    deviceTree.applyOverlays = (prev.callPackage ./apply-overlays-dtmerge.nix { }).applyOverlays;
```

doing `x.applyOverlays = y.applyOverlays` is a bit odd but I'm not sure we want to make `apply-overlays-dtmerge.nix` too different to the original by dropping it's function name. :shrug:

Maybe we can do this?

```suggestion
    deviceTree = prev.deviceTree // (prev.callPackage ./apply-overlays-dtmerge.nix { })
```

The right side should overwrite the left side just fine with the merge

```
nix-repl> { a = "a"; b = "b"; c = "c"; } // { b = "z"; d = "d"; }
{ a = "a"; b = "z"; c = "c"; d = "d"; }
```

---

Or since this module is specifically about modifying applyOverlays maybe we do just change it to return the 1 function & then it'd be

```suggestion
    deviceTree.applyOverlays = (prev.callPackage ./apply-overlays-dtmerge.nix { })
```